mongodbatlas/resource_mongodbatlas_third_party_integration.go

```diff
@@ -55,6 +55,7 @@ func resourceMongoDBAtlasThirdPartyIntegration() *schema.Resource {
 				Required:     true,
 				ForceNew:     true,
 				ValidateFunc: validation.StringInSlice(integrationTypes, false),
+				Deprecated:   "This field type has values (NEW_RELIC, FLOWDOCK) that are deprecated and will be removed in 1.9.0 release ",
```

Comment:
This `Deprecated` field marks the entire argument `type` as deprecated, regardless of its value, which looks misleading to the user:
```
│ Warning: Argument is deprecated
│
│   with module.mongodb_atlas_project[0].module.mongodb_atlas_project.mongodbatlas_third_party_integration.atlas_datadog[0],
│   on <...>/mongodb-atlas-project/common/main.tf line 108, in resource "mongodbatlas_third_party_integration" "atlas_datadog":
│  108:   type = "DATADOG"
│
│ This field type has values (NEW_RELIC, FLOWDOCK) that are deprecated and
│ will be removed in 1.9.0 release
```
Probably, it should be handled by the func instead of a plain string literal here.

Reply:
I opened a new issue for that case: https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1032